kuma/core/models.py

```diff
@@ -34,6 +37,21 @@ def update(self, **kw):
                                           created=False)
 
 
+class IPBan(models.Model):
+    ip = models.GenericIPAddressField()
+    created = models.DateTimeField(default=datetime.now, db_index=True)
+    deleted = models.DateTimeField(null=True, blank=True)
+
+    objects = IPBanManager()
+
+    def delete(self, *args, **kwargs):
+        self.deleted = datetime.now()
+        self.save()
```

Comment:
This mean's we can never delete that db entry, which is not a good pattern. There are apps out there implementing that logical delete pattern much better, I'd strongly recommend using that instead.